miden-lib/src/transaction/inputs.rs

```diff
@@ -25,7 +25,7 @@ impl ToTransactionKernelInputs for PreparedTransaction {
         let account = self.account();
         let stack_inputs = TransactionKernel::build_input_stack(
             account.id(),
-            if account.is_new() { None } else { Some(account.hash()) },
+            account.hash(),
```

Comment:
Here (and in other similar places below), I think we still need to differentiate between new and existing accounts. Specifically, I think the code here should be:
```
if account.is_new() { Digest::default() } else { account.hash() },
```
Basically, for new accounts, `[ZERO; 4]` should go onto the stack.